motionplan/check.go

```diff
@@ -331,8 +337,9 @@ func checkSegments(sfPlanner *planManager, segments []*ik.Segment, lookAheadDist
 			// define State which only houses inputs, pose information not needed since we cannot get arcs from
 			// an interpolating poses, this would only yield a straight line.
 			interpolatedState := &ik.State{
-				Frame:         sfPlanner.frame,
+				Frame:         checkFrame,
 				Configuration: interpConfig,
+				Position:      poseInPath,
```

Comment:
[Q]: Why is this being added back in? IIRC a lot of work was put into CheckPlan so that we do not have to use poses anymore and instead can derive them from the frame's current configuration in terms of its inputs.